signature/src/balrog.rs

```diff
@@ -0,0 +1,366 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use hex;
```

Comment:
Some header comments here would help the reader!

Reply:
I agree! I will add some.